core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -236,11 +237,15 @@ public void watch() {
            * If a directory is created, and we're watching recursively, then register it and its sub-directories.
            */
           if (kind == OVERFLOW) {
-            onEvent(EventType.OVERFLOW, childPath, count, rootPath);
+            boolean isDirectory = isDirectory(pathHashes.get(childPath));
```

Comment:
I believe for `OVERFLOW`, `childPath` can be null, so `pathHashes.get` could throw. It's probably ok to default to `false` if the path is null.

Reply:
isDirectory will return false, if the Hash is null. So not sure I need to do anything?
"return hashCode != null && hashCode.equals(HashCode.EMPTY);"

Reply:
Have you tested it? The issue I think is with `pathHashes.get(null)`.